flytekit/core/context_manager.py

```diff
@@ -274,16 +278,20 @@ def default_deck(self) -> Deck:
 
     @property
     def timeline_deck(self) -> "TimeLineDeck":  # type: ignore
-        from flytekit.deck.deck import TimeLineDeck
+        from flytekit.deck.deck import DeckFields, TimeLineDeck
 
         time_line_deck = None
         for deck in self.decks:
             if isinstance(deck, TimeLineDeck):
                 time_line_deck = deck
                 break
         if time_line_deck is None:
-            time_line_deck = TimeLineDeck("Timeline")
+            if self._timeline_deck is not None:
+                time_line_deck = self._timeline_deck
+            else:
+                time_line_deck = TimeLineDeck(DeckFields.TIMELINE.value)
```

Comment:
I know it's the same, but to be consistent with the other decks:

```suggestion
                time_line_deck = TimeLineDeck("Timeline", auto_add_to_deck=False)
```